GoogleForCreators/web-stories-wp
# Sidebar: New Layers Panel

## Context

Part of editor redesign. 

## Summary

Lifts the layers panel from the inspector pane (which was combined with the design pane to create the new sidebar) and puts it opposite the sidebar as part of the editor's footer. 

## Relevant Technical Choices

Added a few more props to the shared `ToggleButton` that's used for the many popups of the footer to add a variant with copy visible/no icon and prevent tooltip from showing. 

Added `<DirectionAware>` to the footer so that the padding there is consistent and the layers panel wouldn't require additional work to be readable in RTL. I think it looks better now. 

## To-do

- [x] Adjust colors
- [x] Check a11y

Opted against moving all of the layers panel files into their own directory (following the suit of the helpcenter, keyboard nav and checklist) because it makes it harder to grok the updates.

Talked to @agingoldseco about the new toggle design, we're going to do that [later as a nonblocking issue.](https://github.com/GoogleForCreators/web-stories-wp/issues/10891)

The layer panel accessibility is not an issue here, it would be introducing new use patterns should we expose it to the keyboard at this point which would bloat the scope significantly. It was discussed at length 5 months ago in the design slack channel. There are keyboard shortcuts for everything that the layers panel can do. If we decide that now the layers panel should be keyboard accessible that would be its own scope of work and own item on the roadmap. 

## User-facing changes

| LTR | RTL | 
| --- | --- | 
| <img width="1008" alt="Screen Shot 2022-03-11 at 11 23 36 AM" src="https://user-images.githubusercontent.com/10720454/157930585-c3a73edf-5fe5-480d-9709-ce443c3d7bb2.png"> | <img width="1001" alt="Screen Shot 2022-03-11 at 11 24 13 AM" src="https://user-images.githubusercontent.com/10720454/157930620-d0a88572-139d-448e-ac20-65c25546ce85.png"> |

With Meta boxes 
| LTR | RTL | 
| --- | --- | 
| <img width="294" alt="Screen Shot 2022-03-11 at 1 38 26 PM" src="https://user-images.githubusercontent.com/10720454/157959130-c97b40eb-0c4f-4cf5-9028-9aaa8d58b140.png"> | <img width="274" alt="Screen Shot 2022-03-11 at 1 38 37 PM" src="https://user-images.githubusercontent.com/10720454/157959144-e1d25ef6-2180-42d3-9a68-0c86414846c3.png"> |

## Testing Instructions

Use the layers panel, now on the opposite side as the sidebar. 
You should be able to expand and collapse it with cursor and keyboard.
Confirm layer selection and behavior.
Check LTR and RTL. 
Install and activate 'yoast seo' to see meta box show up. 




This PR can be tested by following these steps:

1.


## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #10804
